Detectors/TOF/calibration/testWorkflow/TOFDCSConfigProcessorSpec.h

```diff
@@ -0,0 +1,121 @@
+// Copyright CERN and copyright holders of ALICE O2. This software is
+// distributed under the terms of the GNU General Public License v3 (GPL
+// Version 3), copied verbatim in the file "COPYING".
+//
+// See http://alice-o2.web.cern.ch/license for full licensing information.
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+#ifndef O2_TOF_DCSCONFIGPROCESSOR_H
+#define O2_TOF_DCSCONFIGPROCESSOR_H
+
+/// @file   TOFDCSConfigProcessorSpec.h
+/// @brief  TOF Processor for DCS Configurations
+
+#include "TOFCalibration/TOFFEElightReader.h"
+#include "DetectorsCalibration/Utils.h"
+#include "CommonUtils/MemFileHelper.h"
+#include "CCDB/CcdbObjectInfo.h"
+#include "CCDB/CcdbApi.h"
+
+#include "Framework/DeviceSpec.h"
+#include "Framework/ConfigParamRegistry.h"
+#include "Framework/ControlService.h"
+#include "Framework/WorkflowSpec.h"
+#include "Framework/Task.h"
+#include "Framework/Logger.h"
+
+#include <chrono>
+
+using namespace o2::framework;
+using TFType = uint64_t;
+using HighResClock = std::chrono::high_resolution_clock;
+using Duration = std::chrono::duration<double, std::ratio<1, 1>>;
+
+namespace o2
+{
+namespace tof
+{
+
+class TOFDCSConfigProcessor : public o2::framework::Task
+{
+ public:
+  void init(o2::framework::InitContext& ic) final
+  {
+    mVerbose = ic.options().get<bool>("use-verbose-mode");
+    LOG(INFO) << " ************************* Verbose?" << mVerbose;
+  }
+
+  //---------------------------------------------------------
+
+  void run(o2::framework::ProcessingContext& pc) final
+  {
+    auto configBuff = pc.inputs().get<gsl::span<char>>("inputConfig");
+    auto configFileName = pc.inputs().get<std::string>("inputConfigFileName");
+    auto timer = std::chrono::duration_cast<std::chrono::milliseconds>(HighResClock::now().time_since_epoch()).count();
+    LOG(INFO) << "got input file " << configFileName << " of size " << configBuff.size();
+    mTOFFEElightReader.loadFEElightConfig(configBuff);
+    mTOFFEElightReader.parseFEElightConfig(mVerbose);
+    //auto tfcounter = o2::header::get<o2::framework::DataProcessingHeader*>(pc.inputs().getByPos(0).header)->startTime;
+    sendOutput(pc.outputs(), timer);
+  }
+
+  //---------------------------------------------------------
+
+  void endOfStream(o2::framework::EndOfStreamContext& ec) final
+  {
+  }
+
+ private:
+  void sendOutput(DataAllocator& output, TFType tf)
```

Comment:
did not understand which previous comment you were referring concerning the start time, but I don't see a problem with using now().  In opposite, I would not use `TFType` for something which is a real timestamp: the TFType as TF counter may change in future. Better use `long`

Reply:
Great, I agree, but wanted to be sure. 
`long` is now used.